control/grpc.py

```diff
@@ -1225,6 +1225,13 @@ def namespace_change_load_balancing_group_safe(self, request, context):
                     errmsg = f"{change_lb_group_failure_prefix}: Can't find entry for namespace {request.nsid} in {request.subsystem_nqn}"
                     self.logger.error(errmsg)
                     return pb2.req_status(status=errno.ENOENT, error_message=errmsg)
+                anagrp = ns_entry["anagrpid"]
+                gw_id = self.ceph_utils.get_gw_id_owner_ana_group(self.gateway_pool, self.gateway_group, anagrp)
+                self.logger.debug(f"ANA group of ns#{request.nsid} - {anagrp} is owned by gateway {gw_id}, self.name is {self.gateway_name}")
+                if self.gateway_name != gw_id:
```

Comment:
Shouldn't you check for the case gw_id is None? something like:
`if gw_id and self.gateway_name != gw_id`

Also, the gateway name, at least in the development environment is the container id. Are you sure it's the same as the id you get from ceph?

Reply:
the GW-id should be the same from the ceph.

maybe such condition would be correct
if gw_id == None or self.gateway_name != gw_id :

Reply:
@leonidc this is redundant as the gateway name is not None so in case gw_id is None it will be covered in the condition "self.gateway_name != gw_id". The question is, what should you do in case gw_id is None? Do you want to treat it as if the group is owned by another gateway? My understanding was that in such a case the current gateway can own the group so we need the condition to be as I wrote above.

Reply:
No, if ceph did not find the gw that  is current owner of the group, the  function **get_gw_id_owner_ana_group**  returns None and this should be treat as "owner not found " CLI command cannot be executed in this case